### PR TITLE
Watch external dependent projects correctly

### PR DESCRIPTION
**Pull request by BrennanConroy**
#1912

### CODE REVIEW

File: src/Microsoft.Framework.Runtime/ApplicationHostContext.cs [hunk 1]
```diff
@@ -108,8 +108,11 @@ public class ApplicationHostContext
 
             // Default services
             _serviceProvider.Add(typeof(IApplicationEnvironment), new ApplicationEnvironment(Project, targetFramework, configuration));
-            _serviceProvider.Add(typeof(IFileWatcher), NoopWatcher.Instance);
             _serviceProvider.Add(typeof(ILibraryManager), LibraryManager);
+            if (_serviceProvider.GetService(typeof(IFileWatcher)) == null)
+            {
+                _serviceProvider.Add(typeof(IFileWatcher), NoopWatcher.Instance);
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/ApplicationHostContext.cs]: Add a TryAdd on ServiceProvider

File: src/Microsoft.Framework.Runtime.Compilation.DesignTime/DesignTimeHostCompiler.cs [hunk 2]
```diff
@@ -26,6 +26,7 @@ public DesignTimeHostCompiler(IApplicationShutdown shutdown, IFileWatcher watche
                 foreach (var file in files)
                 {
                     watcher.WatchFile(file);
+                    watcher.WatchProject(file);
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Compilation.DesignTime/DesignTimeHostCompiler.cs]: Check if the file is a project before calling this.

Comment (BrennanConroy) [inline src/Microsoft.Framework.Runtime.Compilation.DesignTime/DesignTimeHostCompiler.cs]: The internals of WatchProject already check if the project is being watched and if not it grabs the project path from file and adds a watcher for the project.

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**